protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -73,7 +74,12 @@ public T parse(InputStream stream) {
           }
         }
         try {
-          return parser.parseFrom(stream);
+          // Pre-create the CodedInputStream so that we can remove the size limit restriction
+          // when parsing.
+          CodedInputStream codedInput = CodedInputStream.newInstance(stream);
+          codedInput.setSizeLimit(Integer.MAX_VALUE);
```

Comment:
This should really be  settable parameter.  That way the default size limit can be maintained (best for general case), can be changed for special cases (cloud bigtable) and make testing the value is enforced easier.